### DISCUSSION TITLE
Feature: how can ABACUS perform metadynamics?

### DISCUSSION THREAD

**Discussion by WHUweiqingzhou** [deepmodeling/abacus-develop#5264] [General]
### Background

Can we perform a metadynamics calculation based on ABACUS? I understand there is no home-built one in ABACUS. Can we calculate it with the help of LAMMPS?

### Describe the solution you'd like

Please support it.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by kirk0830** [comment]
For developer who is interested:
I find Quantum ESPRESSO have the interface with Plumed, therefore mtd can be performed. There is a doc illustrating how to get interfaced with Plumed:
https://github.com/plumed/plumed2/blob/v2.9.2/developer-doc/mdTemplate.md
